logstash-core-event-java/src/main/java/org/logstash/Event.java

```diff
@@ -338,22 +348,33 @@ private Timestamp initTimestamp(Object o) {
             logger.warn("Error parsing " + TIMESTAMP + " string value=" + o.toString());
         }
 
-        tag(TIMESTAMP_FAILURE_TAG);
-        this.data.put(TIMESTAMP_FAILURE_FIELD, o);
-
-        return Timestamp.now();
+        return null;
     }
 
     public void tag(String tag) {
```

Comment:
I think this can be cleaned up a little. what do you think of something like this?

```java
public void tag(String tag) {
    Object existingTags = getUnconvertedField("tags");
    if (existingTags == null) {
        setField("tags", Arrays.asList(tag));
    } else if (existingTags instanceof List) {
        existingTags.add(tag);
    } else if (existingTags instanceof String) {
        List existingTagsAsList = Arrays.asList(existingTags);
        existingTagsAsList.add(tag);
        setField("tags", existingTagsAsList);
    }else {
        // tags is not a List... why did this happen?
        // why do we allow this?
        // how can we prevent this?
        // let's assume that we can't... we should error out?!
    }
}
```

I may be mis-understanding the edge-cases here, though.

Given that both JRuby's and Java's objects implement the `List` and `String` interfaces... I think it is 
safe to use `getUnconvertedField`, and that will be faster (I may be wrong here).